subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/AbstractListChildMap.java

```diff
@@ -122,7 +122,8 @@ protected int findChildIndexWithCommonPrefix(VfsRelativePath targetPath, CaseSen
     }
 
     protected ChildMap<T> withNewChild(int insertBefore, String path, T newChild) {
-        List<Entry<T>> newChildren = new ArrayList<>(entries);
+        List<Entry<T>> newChildren = new ArrayList<>(entries.size() + 1);
+        newChildren.addAll(entries);
```

Comment:
Finally, this doesn't change anything since `addAll` calls `entries.toArray()` which then causes another allocation, so we still have two array allocations instead of one.